exporters/otlp/internal/otlpconfig/envconfig.go

```diff
@@ -71,13 +71,13 @@ func (e *EnvOptionsReader) GetOptionsFromEnv() []GenericOption {
 
 	// Endpoint
 	if v, ok := e.getEnvValue("ENDPOINT"); ok {
-		opts = append(opts, WithEndpoint(v))
+		opts = append(opts, endpointOptions(v, WithEndpoint, withInsecure)...)
```

Comment:
Could this be simplified, and clarified by doing something like:
```
opts = append(opts, 
    WithEndpoint(trimSchema(v)),
    withInsecure(strings.HasPrefix(v, "http://")),
)
```

Reply:
I changed it towards this direction, but with an explicit `WithSecure` function which makes it a little bit more verbose